src/cohort.js

```diff
@@ -0,0 +1,205 @@
+const Twilio = require('../src/sms.js')
+class Cohorts {
+  constructor() {
+    this.cohortList = []
+    this.id = 1
+  }
+
+  addCohort(cohortName) {
+    if (cohortName === '') {
+      throw new Error('Error: Please provide a Cohort name')
+    }
+    for (const cohort of this.cohortList) {
+      const cohortNameList = Object.keys(cohort)[0]
+      if (cohortNameList === cohortName) {
+        throw new Error('Error: The name already exists')
+      }
+    }
+    const newCohort = { [cohortName]: [] }
+    this.cohortList.push(newCohort)
+    return this.cohortList
+  }
+
+  searchByName(cohortName) {
+    if (this.cohortList.length === 0) {
+      throw new Error('ERROR: Cohort List is empty')
+    }
+
+    const targetCohort = this.cohortList.find((cohort) => {
+      const cohortNameList = Object.keys(cohort)[0]
+      return cohortNameList === cohortName
+    })
+
+    if (!targetCohort) {
+      throw new Error('Error: Cohort name does not exist')
+    }
+
+    return targetCohort
+  }
+
+  removeByName(cohortName) {
+    if (this.cohortList.length === 0) {
+      throw new Error('ERROR: Cohort List is empty')
+    }
+
+    for (const cohort of this.cohortList) {
+      const cohortNameList = Object.keys(cohort)[0]
+      if (cohortNameList === cohortName) {
+        const index = this.cohortList.indexOf(cohort)
+        this.cohortList.splice(index, 1)
+        console.log(this.cohortList)
+        return this.cohortList
+      } else if (cohortNameList !== cohortName) {
+        throw new Error('Error: Cohort name does not exist')
+      }
+    }
+  }
+
+  addStudentToCohort(
```

Comment:
I used just the name and ID to add new student - perhaps I should have also included all the rest of the info too